.github/workflows/my-ci-pipeline.yml

```diff
@@ -0,0 +1,33 @@
+name: Continuous Integration 
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+      - 'README.md'
+  pull_request:
+    branches: [main]
```

Comment:
Note that this means any PR that targets main; so it's good in that it will run our tests once we're ready to merge & raising a PR, but it won't target in-progress branches which we often want.
Allowing Push/PR to both execute on all branches (by just removing those sections) might be nicer, so that we get tests running often, but see my comment below on limiting the publishing step